app/moves/controllers/list-by-status.js

```diff
@@ -1,16 +1,39 @@
 const presenters = require('../../../common/presenters')
+const i18n = require('../../../config/i18n')
+const filters = require('../../../config/nunjucks/filters')
+
+const tableConfig = [
```

Comment:
This feels a lot like our [`presenters`](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/tree/master/common/presenters) pattern of taking some data from a service or API request and transforming it into how a specific component expects to receive that data. 

Maybe it could be re-worked to continue to use that pattern consistently?

Reply:
Ah, sorry, hadn't seen you are using presenters. Maybe it could be combined to use them like they're being used to build [other components](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/master/common/presenters/person-to-summary-list-component.js)?

Reply:
The issue here, as mentioned in the comment PR, is that I am trying to avoid tying the presenters of the rows and table to a particular type of object. Here's moves, but soon those presenters will be used for allocations as well.
If I used two series of presenters there would be a lot of duplication. I could use a mixin-like pattern to have the main presenters methods in common and two different configs, something like
-- allocationToTableRow / allocationToTableHead
-- moveToTableRow / moveToTableHead
if it feels cleaner.

Reply:
However, on second thought, no matter how I change it feels like a complication. I remember you telling me that the presenter should be kept as simple as possible and that any `map` should be invoking it instead of containing it. Therefore, I have kept separated row and heading; if I create a general "moves table presenter" and another "allocation table presenter" that invokes the two presenters with the config, I go against this, not to mention that it becomes clunkier.

What's the main issue with keeping the table config within the controller? It seems acceptable to instruct a generic presenter on what goes where, no?

Reply:
Yes, I suppose in this instance it's tricky to just call `.map()` on a presenter because it requires all items in the collection.

But I think it feels a little cleaner to have one presenter that transforms a list of moves to a table format. Have a look at what we do with `movesByLocation`. The presenter doesn't necessarily have to transform to a component. 

And with allocations, I'd say cross that bridge when it comes to it. An `allocation` will also have a different structure of a `move` anyway and having config and passing two separate values to build the table could over complicate it.

For this change I'd look at just creating a presenter to take a collection of moves and output in the format the `govukTable` components wants.

That way the controller remains simple too, which is a pattern we've tried to adopt where possible.